### ISSUE TITLE
optimize attribute reading in models/read_models??

### ISSUE THREAD

**Issue by kellyredding** [redding/mr#217]
(from a tweet)
ActiveRecord protip: I went from 3800 ops/sec to 24000 ops/sec by switching from attributes[‘name’] to read_attribute(‘name’). (https://twitter.com/mperham/status/466009719727869952) 

We should bench and see if there are any noticeable improvements with this.

**Comment by jcredding** [comment]
@kellyredding - I'm closing this. I tried this out and didn't see any performance improvement. I think most of the improvement that he was seeing was because he was using `attributes` which built the entire hash of all attributes values and `read_attribute` returned a single value. 

For models we are using the default reader which apparently is the same as using `read_attribute` unless you override the reader method on your record. For read models we are using the hash accessor `[]` which is only returning a single value and is also apparently the same as `read_attribute` and the default reader. I hacked in `read_attribute` anyway and ran the bench scripts and didn't see any noticeable change so I don't think there's any performance gains to be made here.
